akka-docs/src/main/paradox/persistence-typed.md

```diff
@@ -1,32 +1,179 @@
+
+The next example shows how to define different behavior based on the current `State`. It is an actor that
+represents the state of a blog post. Before a post is started the only command it can process is to `AddPost`. Once it is started
+then it we can look it up with `GetPost`, modify it with `ChangeBody` or publish it with `Publish`.
+
+The state is captured by:
+
+Scala
+:  @@snip [InDepthPersistentActorSpec.scala]($akka$/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/InDepthPersistentActorSpec.scala) { #state }
+
+The commands (only a subset are valid depending on state):
+
+Scala
+:  @@snip [InDepthPersistentActorSpec.scala]($akka$/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/InDepthPersistentActorSpec.scala) { #commands }
+
+The command handler to process each command is decided by a `CommandHandler.byState` command handler, 
+which is a function from `State => CommandHandler`:
+
+Scala
+:  @@snip [InDepthPersistentActorSpec.scala]($akka$/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/InDepthPersistentActorSpec.scala) { #by-state-command-handler }
+
+This can refer to many other `CommandHandler`s e.g one for a post that hasn't been started:
+
+Scala
+:  @@snip [InDepthPersistentActorSpec.scala]($akka$/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/InDepthPersistentActorSpec.scala) { #initial-command-handler }
+
+And a different `CommandHandler` for after the post has been added:
+
+Scala
+:  @@snip [InDepthPersistentActorSpec.scala]($akka$/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/InDepthPersistentActorSpec.scala) { #post-added-command-handler }
+
+The event handler is always the same independent of state. The main reason for not making the event handler 
+part of the `CommandHandler` is that all events must be handled and that is typically independent of what the 
+current state is. The event handler can of course still decide what to do based on the state if that is needed.
+
+Scala
+:  @@snip [InDepthPersistentActorSpec.scala]($akka$/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/InDepthPersistentActorSpec.scala) { #event-handler }
+
+And finally the behavior is created from the `byState` command handler:
+
+Scala
+:  @@snip [InDepthPersistentActorSpec.scala]($akka$/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/InDepthPersistentActorSpec.scala) { #behavior }
+
+## Serialization
+
+The same @ref:[serialization](serialization.md) mechanism as for untyped 
+actors is also used in Akka Typed, also for persistent actors. When picking serialization solution for the events 
+you should also consider that it must be possible read old events when the application has evolved. 
+Strategies for that can be found in the @ref:[schema evolution](persistence-schema-evolution.md).
+
+## Recovery
+
+For executing side effects after a recovery is completed use `onRecoveryCompleted` on a `PersistenetActor` e.g
```

Comment:
Since it is strongly discouraged to perfom side effects in `applyEvent` (didn't we change it to that, it's still event handler???), side effects should be performec once recovery has completed in the `onRecoveryCompleted` callback

?

Reply:
I like that wording